lib/fluent/agent.rb

```diff
@@ -15,7 +15,6 @@
 #
 
 require 'fluent/configurable'
-require 'fluent/engine'
```

Comment:
Removing this `require` looks good, but `Engine` is referred below to get logger.
I think we need to fix `Agent` to get logger in constructor, and to pass it from Engine when RootAgent is created in `engine.rb`.